lib/chef/provider/git.rb

```diff
@@ -313,18 +327,7 @@ def run_options(run_opts = {})
           # Certain versions of `git` misbehave if git configuration is
           # inaccessible in $HOME. We need to ensure $HOME matches the
           # user who is executing `git` not the user running Chef.
-          env["HOME"] =
-            begin
-              require "etc" unless defined?(Etc)
-              case new_resource.user
-              when Integer
-                Etc.getpwuid(new_resource.user).dir
-              else
-                Etc.getpwnam(new_resource.user.to_s).dir
-              end
-            rescue ArgumentError # user not found
-              raise Chef::Exceptions::User, "Could not determine HOME for specified user '#{new_resource.user}' for resource '#{new_resource.name}'"
```

Comment:
This decoration of the exception has been lost in the refactoring since `get_dir` isn't rescuing and re-raising any more.